File: Easy/classconsorsuale/meta_classconsorsualedefaultview/meta_classconsorsualedefaultview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_classconsorsualedefaultview
{
    public class Meta_classconsorsualedefaultview : Meta_easydata 
	{
        public Meta_classconsorsualedefaultview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "classconsorsualedefaultview") {
				Name = "Classi di concorso MIUR";
			EditTypes.Add("default");
			ListingTypes.Add("default");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idclassconsorsuale"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "default": {
						DescribeAColumn(T, "idclassconsorsuale", "Codice", nPos++);
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "classconsorsuale_description", "Descrizione", nPos++);
						DescribeAColumn(T, "classconsorsuale_active", "Attivo", nPos++);
						DescribeAColumn(T, "classconsorsuale_ambitodisci", "Ambito Disciplinare", nPos++);
						DescribeAColumn(T, "classconsorsuale_corr2592017", "Corrispondenza", nPos++);
						DescribeAColumn(T, "classconsorsuale_normativa", "Normativa", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "default": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/classconsorsuale/meta_classconsorsualedefaultview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_classconsorsualedefaultview() {
        MetaData.apply(this, ["classconsorsualedefaultview"]);
        this.name = 'meta_classconsorsualedefaultview';
    }

    meta_classconsorsualedefaultview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_classconsorsualedefaultview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'default':
						this.describeAColumn(table, 'title', 'Codice', null, 1000, 50);
						this.describeAColumn(table, 'classconsorsuale_description', 'Descrizione', null, 2000, 512);
						this.describeAColumn(table, 'classconsorsuale_active', 'Attivo', null, 3000, null);
						this.describeAColumn(table, 'classconsorsuale_ambitodisci', 'Ambito Disciplinare', null, 4000, 50);
						this.describeAColumn(table, 'classconsorsuale_normativa', 'Normativa', null, 5000, 50);
						this.describeAColumn(table, 'classconsorsuale_corr2592017', 'Corrispondenza', null, 6000, 50);
						this.describeAColumn(table, 'classconsorsuale_tipoente', 'Tipologia di ente', null, 10000, null);
//$objCalcFieldConfig_default$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idclassconsorsuale"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "default": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('classconsorsualedefaultview', new meta_classconsorsualedefaultview('classconsorsualedefaultview'));

	}());
